### PR TITLE
SOC-1939 | Tracking for upvoting and undoing upvoting

### PR DESCRIPTION
**Pull request by bkoval**
## Links

* http://wikia-inc.atlassian.net/browse/JIRA-1939

## Description

Tracking for upvotes / downvotes

## Reviewers

@rafalkalinski @kvas-damian

### CODE REVIEW

File: front/main/app/mixins/discussion-route-upvote.js [hunk 1]
```diff
@@ -32,6 +33,12 @@ export default Ember.Mixin.create({
 				url: M.getDiscussionServiceUrl(`/${Ember.get(post, 'siteId')}/votes/post/${Ember.get(post, 'id')}`),
 				success: (data) => {
 					Ember.set(post, 'upvoteCount', data.upvoteCount);
+
+					if (hasUpvoted) {
+						track(trackActions.UndoUpvotePost);
+					} else {
+						track(trackActions.UpvotePost);
+					}
```
Comment (rafalkalinski) [inline front/main/app/mixins/discussion-route-upvote.js]: Maybe:
`const actionToTrack = hasUpvoted ? 'UndoUpvotePost' : 'UpvotePost';`
`track(trackActions[actionToTrack]);`

**Comment by rafalkalinski** [comment]
LGTM

File: front/main/app/mixins/discussion-route-upvote.js (hunk 1 above)
Comment (bkoval) [inline front/main/app/mixins/discussion-route-upvote.js]: It's possible, but an if statement doesn't require you to create an additional local variable. Also, it seems a bit more readable to me.

File: front/main/app/utils/discussion-tracker.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ import {track as mercuryTrack} from 'common/utils/track';
 const trackActions = {
 	PostCreate: 'PostCreate',
 	ReplyCreate: 'ReplyCreate',
+	UndoUpvotePost: 'UndoUpvotePost',
+	UpvotePost: 'UpvotePost',
```
Comment (kvas-damian) [inline front/main/app/utils/discussion-tracker.js]: Those actions suggest tracking only posts. What about replies? Are they handled separately?

Comment (bkoval) [inline front/main/app/utils/discussion-tracker.js]: It's intentional. We track upvotes on posts and replies using the same action name ( as mobile apps do -> https://docs.google.com/spreadsheets/d/1Gk1iqWkpjjVgZNiooIfYUP2X5MunFkcDt2RHQhK7b7A/edit#gid=3 )

Comment (kvas-damian) [inline front/main/app/utils/discussion-tracker.js]: Let it be! 
It's ok for me while it's accepted by Product Owner.

**Comment by kvas-damian** [comment]
LGTM

**Comment by bkoval** [comment]
Closing in favor of #2217

**Pull request closed without merging**